File: php/tests/Conformance/AmendFixture/Table/AmendWidget/V0001_Create.php
<?php

declare(strict_types=1);

namespace SimpleOrm\Tests\Conformance\AmendFixture\Table\AmendWidget;

use SimpleOrm\Migrations\TableActions;
use SimpleOrm\Migrations\TableMigration;
use SimpleOrm\Tests\Conformance\AmendFixture\AmendWidget;

final class V0001_Create extends TableMigration
{
    public function entityClass(): string
    {
        return AmendWidget::class;
    }

    public function action(TableActions $actions): void
    {
        $actions->sql('create table amend_widgets (id INTEGER PRIMARY KEY, name TEXT NOT NULL) STRICT');
    }
}

File: go/orm/internal/conformance/amendfixture/Table/AmendWidget/V0001_Create.go
// Package amendwidget holds the amend_widgets table's fixture migration steps.
package amendwidget

import (
	"github.com/kosmassp/SimpleOrm/go/orm"
	"github.com/kosmassp/SimpleOrm/go/orm/internal/conformance/amendfixture/models"
)

// V0001_Create creates the fixture table.
type V0001_Create struct {
	orm.TableMigration[models.AmendWidget]
}

func (V0001_Create) Action(actions *orm.TableActions) {
	actions.SQL("create table if not exists amend_widgets (id INTEGER PRIMARY KEY, name TEXT NOT NULL) STRICT")
}
